src/main/webapp/app/modules/canvas/visualization/visualization-modal/visualization-edit-modal/visualization-edit-modal.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect } from 'react';
+import { View, Flex, Dialog, Heading, Divider, Content, Text, ButtonGroup, Button, DialogContainer } from '@adobe/react-spectrum';
+import { IRootState } from 'app/shared/reducers';
+import { connect } from 'react-redux';
+import './visualization-edit-modal.scss';
+import { RouteComponentProps, useHistory } from 'react-router-dom';
+import VisualizationProperties from 'app/modules/canvas/visualization/visualization-properties/visualization-properties';
+import VisualizationSettings from 'app/modules/canvas/visualization/visualization-settings/visualization-settings';
+import { Translate } from 'react-jhipster';
+import {
+  getEntity as getVisualmetadataEntity,
+  updateEntity as updateVisualmetadataEntity,
+} from 'app/entities/visualmetadata/visualmetadata.reducer';
+import { getDatasourcesFeaturesEntities as getfeatureEntities } from 'app/entities/feature/feature.reducer';
+import { getEntity as getViewEntity } from 'app/entities/views/views.reducer';
+import { VisualMetadataContainerUpdate } from '../../util/visualmetadata-container.service';
+import { setVisual } from '../../util/VisualDispatchService';
+export interface IVisualizationEditModalProps1
+  extends StateProps,
+    DispatchProps,
+    RouteComponentProps<{ id: string; viewId: string; visualizationId: string }> {}
+
+const VisualizationEditModal = (props: IVisualizationEditModalProps1) => {
+  const history = useHistory();
```

Comment:
no need to use this. you can do props.history.push

Reply:
this file is not in use now
deleting this file